tools/dds/dds-server/lrs-device-watcher.cpp

```diff
@@ -28,15 +28,11 @@ void lrs_device_watcher::run( std::function< void( rs2::device ) > add_device_cb
     // Register to LRS device change callback.
     // For each device added, call callback and store it in a list
     // For each device removed, call callback and remove it from the list
-    std::weak_ptr<lrs_device_watcher> weak_dev_watcher = shared_from_this();
+    std::weak_ptr<void> weak_dev_watcher(_alive);
     _ctx.set_devices_changed_callback(
         [this, add_device_cb, remove_device_cb, weak_dev_watcher]( rs2::event_information & info ) {
-            if( _running )
+            if( weak_dev_watcher.lock() )
```

Comment:
Now that I see this, we need to assign it to a shared_ptr. I.e., the shared_ptr needs to point to the `_rs_device_list`...
Let's talk again?

Reply:
Changes to shared_ptr<std::vector<rs2::device>>